### PR TITLE
Merge SERVICE_STATE_MANAGEMENT_TASK to SERVICE_ORDER.

### LINKED ISSUE #2026: Merge SERVICE_STATE_MANAGEMENT_TASK to SERVICE_ORDER
In order to consolidate DB structure, we will make SERVICE_ORDER table as the master table to hold all changes requested on the service. 

- [ ] Remove SERVICE_STATE_MANAGEMENT_TASK table completely. 
- [ ] Record all changes only to SERVICE_ORDER. 
- [ ] Extend ServiceOrderType enum with SERVICE_START, SERVICE_STOP and SERVICE_RESTART
- [ ] Update ServiceStateManageApi to use SERVICE_ORDER table for all changes. 
- [ ] Delete services deleteManagementTaskByTaskId, deleteManagementTasksByServiceId, listServiceStateManagementTasks, getManagementTaskDetailsByTaskId
- [ ] For start, restart, stop - return ServiceOrder object. Ensure exceptions also contain the serviceId and OrderId
- [ ] Remove latestRunningManagementTask and latestModificationAudit fields in DEPLOY_SERVICE

### PR DESCRIPTION
**Pull request by Alice1319**
fixes #2026 

stop successfully:
image

image

start successfully:
image
image

restart successfully:
image
image

start exception:
image

restart exception:
image

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java [hunk 1]
```diff
@@ -57,7 +55,7 @@ public class GetCspInfoFromRequest {
     @Resource
     private DatabaseServiceMigrationStorage serviceMigrationStorage;
     @Resource
-    private DatabaseServiceStateManagementTaskStorage managementTaskStorage;
+    private DatabaseServiceOrderStorage serviceOrderStorage;
     @Resource
     private DatabaseServiceOrderStorage serviceOrderTaskStorage;
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java:60]: Repeatedly important the same bean of `DatabaseServiceOrderStorage`, please delete one.

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java [hunk 2]
```diff
@@ -244,8 +242,8 @@ public Csp getCsFromWorkflowTaskId(String taskId) {
      */
     public Csp getCspFromManagementTaskId(String managementTaskId) {
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java:243]: Delete this method and use `getCspFromServiceOrderId(String orderId)`.

**Comment by WangLiNaruto** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java (hunk 1 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java:58]: thank u for reminding this, fixed.

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java (hunk 2 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java]: deleted.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java [hunk 3]
```diff
@@ -66,32 +60,37 @@ public class ServiceStateManager {
...
+        serviceOrder.setServiceId(newTask.getServiceId());
+        serviceOrder.setOrderId(newTask.getOrderId());
+        return serviceOrder;
     }
 
-    private ServiceStateManagementTaskEntity createNewManagementTask(
-            UUID serviceId, ServiceStateManagementTaskType taskType) {
-        ServiceStateManagementTaskEntity newTask = new ServiceStateManagementTaskEntity();
-        newTask.setTaskId(UUID.randomUUID());
+    private ServiceOrderEntity createNewManagementTask(
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java:77]: use the methods available in createServiceOrderTask to create service_order entities

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java [hunk 4]
```diff
@@ -119,22 +118,26 @@ private void asyncStartService(ServiceStateManagementTaskEntity taskEntity,
...
         taskExecutor.execute(() -> asyncStopService(newTask, plugin, stopRequest, service));
-        return newTask.getTaskId();
+        ServiceOrder serviceOrder = new ServiceOrder();
+        serviceOrder.setServiceId(newTask.getServiceId());
+        serviceOrder.setOrderId(newTask.getOrderId());
+        return serviceOrder;
     }
 
-    private void asyncStopService(ServiceStateManagementTaskEntity taskEntity,
+    private void asyncStopService(ServiceOrderEntity taskEntity,
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java:135]: please correct the argument names everywhere.. taskEntity is not the correct one in this case.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java (hunk 3 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java:77]: fixed.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java (hunk 4 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java:135]: use serviceOrderTaskEntity instead.

File: modules/database/src/main/java/org/eclipse/xpanse/modules/database/service/DeployServiceEntity.java [hunk 5]
```diff
@@ -138,7 +138,7 @@ public class DeployServiceEntity extends CreateModifiedTime {
     @ToString.Exclude
     private ServiceConfigurationEntity serviceConfigurationEntity;
 
-    @OneToMany(mappedBy = "deployServiceEntity", orphanRemoval = true)
+    @OneToMany(mappedBy = "deployServiceEntity")
```
Comment (swaroopar) [inline modules/database/src/main/java/org/eclipse/xpanse/modules/database/service/DeployServiceEntity.java:141]: why is this removed? we must remove orphan entities here.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**